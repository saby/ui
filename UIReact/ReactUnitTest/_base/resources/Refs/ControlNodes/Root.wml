<div>
    <div>update controlNode events: {{wasChildrenControlNodeUpdate}}</div>
    <ReactUnitTest._base.resources.Refs.ControlNodes.ItemsWrapper on:itemClick="itemClickHandler()"/>
</div>
