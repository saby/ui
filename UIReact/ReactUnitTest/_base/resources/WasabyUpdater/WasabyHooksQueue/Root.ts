import { Control } from 'UICore/Base';
import template = require('wml!ReactUnitTest/_base/resources/WasabyUpdater/WasabyHooksQueue/Root');

export default class WasabyHooksQueueRoot extends Control {
    protected _template: unknown = template;
}
