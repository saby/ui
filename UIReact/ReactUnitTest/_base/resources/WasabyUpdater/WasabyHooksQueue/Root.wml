<div>
   <div>WasabyHooksQueueRoot</div>
   <ReactUnitTest._base.resources.WasabyUpdater.WasabyHooksQueue.NumberZero />
   <ReactUnitTest._base.resources.WasabyUpdater.WasabyHooksQueue.NumberZeroNeighbor />
</div>
