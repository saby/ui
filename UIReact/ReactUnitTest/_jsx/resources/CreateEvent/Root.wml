<div>
    <ReactUnitTest._jsx.resources.CreateEvent.ReactControl />
</div>
