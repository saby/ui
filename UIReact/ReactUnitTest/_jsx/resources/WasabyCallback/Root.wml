<div>
    <div>
        {{result}}
    </div>
    <ReactUnitTest._jsx.resources.WasabyCallback.ReactControl on:result="resultHandlerEvent()" isEvent="{{true}}" />
    <ReactUnitTest._jsx.resources.WasabyCallback.ReactControl on:result="resultHandlerNoEvent()" isNoEvent="{{true}}"/>
    <ReactUnitTest._jsx.resources.WasabyCallback.ReactControl on:result="resultHandlerWithVal('myVal', 'myVal1')" isEmpty="{{true}}" />
</div>
