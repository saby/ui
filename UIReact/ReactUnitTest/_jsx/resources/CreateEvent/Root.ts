import { Control, TemplateFunction } from 'UI/Base';
/* eslint-disable no-console */
// @ts-ignore
import * as template from 'wml!ReactUnitTest/_jsx/resources/CreateEvent/Root';

export default class Root extends Control {
    protected _template: TemplateFunction = template;
}
