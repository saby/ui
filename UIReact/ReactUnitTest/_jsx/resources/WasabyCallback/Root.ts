import { Control, TemplateFunction } from 'UI/Base';
/* eslint-disable no-console */
// @ts-ignore
import * as template from 'wml!ReactUnitTest/_jsx/resources/WasabyCallback/Root';

export default class Root extends Control {
    protected _template: TemplateFunction = template;
    protected result: string = 'init';

    resultHandlerEvent(e, val, val1) {
        this.result = `${e.type} ${val} ${val1}`;
    }

    resultHandlerNoEvent(val, val1) {
        this.result = `${val} ${val1}`;
    }

    resultHandlerWithVal(_, val, val1) {
        this.result = `${val} ${val1}`;
    }
}
