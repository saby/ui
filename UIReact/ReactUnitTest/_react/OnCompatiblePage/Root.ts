import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/_react/OnCompatiblePage/Root';

export default class Root extends Control {
    _template: TemplateFunction = template;
    _showReactControl: boolean = false;
    _showWs3Control: boolean = false;

    toggleReactControl(): void {
        this._showReactControl = !this._showReactControl;
    }

    toggleWs3Control(): void {
        this._showWs3Control = !this._showWs3Control;
    }
}
