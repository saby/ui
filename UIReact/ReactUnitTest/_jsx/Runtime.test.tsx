/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode, render } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Root from './Runtime/Root';
const RootTemplate = requirejs('wml!ReactUnitTest/_jsx/Runtime/RootTemplate');

describe('Подмена React.createElement для совместимости с васаби', () => {
    let container: HTMLDivElement;

    beforeEach(() => {
        /*
     _afterMount и _afterUpdate зовутся в отдельной таске, чтобы браузер мог отрисовать кадр.
     Чтобы не делать тесты асинхронными, мы просто мокнем таймеры и сами будем управлять временем.
      */
        jest.useFakeTimers();
        container = document.createElement('div');
        document.body.appendChild(container);
        jest.spyOn(window, 'requestAnimationFrame').mockImplementation(
            setTimeout
        );
    });

    afterEach(() => {
        jest.useRealTimers();
        unmountComponentAtNode(container);
        container.remove();
        container = null;
        jest.restoreAllMocks();
    });

    it('Вставка wasaby-контрола с генерацией React.createElement', () => {
        act(() => {
            const obj = {};
            render(<div><Root {...obj} key={123} value={123} className={'hello'}/></div>, container);
        });

        expect(container).toMatchSnapshot();
    });
    it('Вставка wasaby-шаблона с генерацией React.createElement', () => {
        act(() => {
            const obj = {};
            render(<div><RootTemplate {...obj} key={123} value={123} className={'hello'}/></div>, container);
        });

        expect(container).toMatchSnapshot();
    });


});
