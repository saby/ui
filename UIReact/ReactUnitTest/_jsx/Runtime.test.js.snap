// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Подмена React.createElement для совместимости с васаби Вставка wasaby-контрола с генерацией React.createElement 1`] = `
<div>
  <div>
    <div
      class="hello"
      tabindex="0"
    >
      123
    </div>
  </div>
</div>
`;

exports[`Подмена React.createElement для совместимости с васаби Вставка wasaby-шаблона с генерацией React.createElement 1`] = `
<div>
  <div>
    <div
      class="hello"
      tabindex="0"
    >
      123
    </div>
  </div>
</div>
`;
